src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -127,6 +127,46 @@ public async Task<IActionResult> GetRecommendations(string sessionId)
             return Ok(output);
         }
 
+        /// <summary>
+        /// Sets a value for a list of option setting items on the selected recommendation.
+        /// Option setting updates are provided as Key Value pairs with the Key being the JSON path to the leaf node.
+        /// Only primitive data types are supported for Value updates. The Value is a string value which will be parsed as its corresponding data type.
+        /// </summary>
+        [HttpPut("session/<sessionId>/settings")]
+        [SwaggerOperation(OperationId = "SetConfigSettings")]
+        [SwaggerResponse(200, type: typeof(SetConfigSettingsOutput))]
+        [Authorize]
+        public IActionResult SetConfigSettings(string sessionId, [FromBody] SetConfigSettingsInput input)
+        {
+            var state = _stateServer.Get(sessionId);
+            if (state == null)
+            {
+                return NotFound($"Session ID {sessionId} not found.");
+            }
+
+            if (state.SelectedRecommendation == null)
+            {
+                return NotFound($"A deployment target is not set for Session ID {sessionId}.");
+            }
+
+            var output = new SetConfigSettingsOutput();
+
+            foreach (var updatedSetting in input.UpdatedSettings)
+            {
+                try
+                {
+                    var setting = state.SelectedRecommendation.GetOptionSetting(updatedSetting.Key);
+                    setting.SetValueOverride(updatedSetting.Value);
+                }
+                catch (Exception ex)
+                {
+                    output.FailedConfigUpdates.Add(updatedSetting.Key, ex.Message);
+                }
+            }
+
+            return Ok(output);
```

Comment:
Should the response contain the current state of the settings? Some settings default values could have changed and the state of updatablity can change as well. We can either return them here along with the errors or the UI has to make a separate call get the settings again.

Reply:
I thought about this after reading the document and did a small research and the convention is just to do the update for PUT requests. To get the new state, you should do another GET. That is what I am doing from the toolkit. After a PUT, I do another GET to check the items.